### PR TITLE
[OptApp] Update filter interfaces

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
As discussed on #12158 , This changes the `Filter` interface.

Please check whether the comments also make sense.

**🆕 Changelog**
- Change `Filter` interface class.

### CODE REVIEW

File: applications/OptimizationApplication/python_scripts/filtering/filter.py [hunk 1]
```diff
@@ -74,43 +74,62 @@ def Finalize(self) -> None:
...
-            ContainerExpressionTypes: Filtered output field.
+            ContainerExpressionTypes: Mesh-independent gradient in the control space.
         """
         pass
 
-    def UnfilterField(self, filtered_field: ContainerExpressionTypes) -> ContainerExpressionTypes:
-        """Get the unfiltered field from a filtered field
+    @abc.abstractmethod
+    def BackwardFilterIntegratedField(self, physical_mesh_dependent_gradient_field: ContainerExpressionTypes) -> ContainerExpressionTypes:
+        """Returns control space gradient from mesh-dependent physical space gradient.
```
Comment (RezaNajian) [inline applications/OptimizationApplication/python_scripts/filtering/filter.py]: i would add returns mesh-independent control space gradient from ....

File: applications/OptimizationApplication/python_scripts/filtering/filter.py [hunk 2]
```diff
@@ -74,43 +74,62 @@ def Finalize(self) -> None:
...
         Returns:
-            ContainerExpressionTypes: Filtered output field.
+            ContainerExpressionTypes: Physical field.
         """
         pass
 
     @abc.abstractmethod
-    def FilterIntegratedField(self, unfiltered_integrated_field: ContainerExpressionTypes) -> ContainerExpressionTypes:
-        """Filter the input integrated field.
+    def BackwardFilterField(self, physical_mesh_independent_gradient_field: ContainerExpressionTypes) -> ContainerExpressionTypes:
```
Comment (RezaNajian) [inline applications/OptimizationApplication/python_scripts/filtering/filter.py:92]: here as well returns mesh-independent control space gradient

**Review by RezaNajian — DISMISSED** [review]

File: applications/OptimizationApplication/python_scripts/filtering/filter.py (hunk 2 above)
Comment (sunethwarna) [reply applications/OptimizationApplication/python_scripts/filtering/filter.py:92]: done :)

File: applications/OptimizationApplication/python_scripts/filtering/filter.py (hunk 1 above)
Comment (sunethwarna) [reply applications/OptimizationApplication/python_scripts/filtering/filter.py]: done :)

**Review by RezaNajian — APPROVED** [review]

**Pull request merged**